tidysic/parser/tree.py

```diff
@@ -25,8 +32,46 @@ def _parse(self) -> None:
         """
         for path in self._root.iterdir():
             if path.is_dir():
-                self.children.add(Tree(path))
+                child = Tree(path)
+                if child.common_tags is not None:
+                    self.children.add(child)
+                else:
+                    self.clutter_files.add(TaggedFile(path))
             elif AudioFile.is_audio_file(path):
                 self.audio_files.add(AudioFile(path))
             else:
-                self.clutter_files.add(path)
+                self.clutter_files.add(TaggedFile(path))
+
+        self._tag_clutter()
+
+    def _tag_clutter(self):
+        """
+        Tags non-audio files with the tags common to all audio files in the same
+        directory and subdirectories.
+        """
+        self._find_common_tags()
+        self._apply_common_tags_to_clutter()
+
+    def _find_common_tags(self):
+        """
+        Finds the common tags shared by the given tagged objects.
+        """
+        children_tags = [
```

Comment:
Unlikely but, this could be a set as multiple children may have the same common tags (e.g., if you are dealing with files that do not have any title tag).

This would also be more consistent with the Tree attributes that are also `set`.

Reply:
`Taggable`s cannot be put in a set because they are unhashable. Making them hashable would require them to be `frozen` dataclasses, which we cannot accommodate.